### PR TITLE
docs: cleanups, formatting, doctests

### PR DESCRIPTION
**Pull request by Xiretza**
I went through the docs and improved what I could find. `utils` and `Client` got some doctests, everything else basically requires a working server and authentication, making it hard to write doctests.

### CODE REVIEW

File: src/online_managers.rs [hunk 1]
```diff
@@ -262,15 +263,11 @@ impl<'a> Authenticator<'a> {
         }
     }
 
-    pub fn is_etebase_server(&self) -> Result<bool> {
+    pub fn validate_server(&self) -> Result<()> {
         let url = self.api_base.join("is_etebase/")?;
         let res = self.client.get(url.as_str())?;
-        if res.status() == 404 {
-            return Ok(false);
-        }
```
Comment (Xiretza) [inline src/online_managers.rs:270]: What is the reason behind this? Why is a 404 different from any other error? If that's really an important distinction, the method should probably return a custom type that makes it more descriptive.

Comment (Xiretza) [reply src/online_managers.rs]: Ah, that makes sense. I changed it so the functionality stays the same, but the method is still moved to a more obvious place and has a more descriptive name.

File: src/http_client/mod.rs [hunk 2]
```diff
@@ -63,6 +95,23 @@ impl Client {
...
+    /// ```
+    /// use etebase::Client;
+    ///
+    /// let invalid_client = Client::new("ExampleClient", "https://example.com").unwrap();
+    /// assert!(!invalid_client.has_valid_server().unwrap());
+    ///
+    /// let valid_client = Client::new("ExampleClient", "https://api.etebase.com/").unwrap();
+    /// assert!(valid_client.has_valid_server().unwrap());
+    /// ```
+    pub fn has_valid_server(&self) -> Result<bool> {
```
Comment (tasn) [inline src/http_client/mod.rs:111]: Maybe is_valid_server? I liked that better.

Comment (Xiretza) [reply src/http_client/mod.rs:111]: To me that would imply "is this object (of type `Client`) a valid server?", which is a bit of a nonsensical question. What do you think of `is_server_valid()`?

Comment (tasn) [reply src/http_client/mod.rs:111]: Sounds good.

Comment (Xiretza) [reply src/http_client/mod.rs]: Done!

**Review by tasn — APPROVED** [review]

**Pull request merged**